group_dynamic throws expection: implementation error groups are out of bounds; please open an issue

#### What language are you using?

Python

#### Which feature gates did you use?


#### What version of polars are you using?
0.13.5

#### What operating system are you using polars on?

Ubuntu 20.04

#### What language version are you using
python 3.8

#### Describe your bug.

group_dynamic throws a expection when the 'period' value and absolute value of a negative offset is bigger then the 'every' parameter value

#### What are the steps to reproduce the behavior?



```python
import polars as pl
from datetime import datetime

df = pl.from_pandas(pd.DataFrame(data=[
        ['Mali', 'typ1',datetime(2020,1,1), 'actor1', 0],
        ['Mali', 'typ1',datetime(2020,3,1), 'actor2', 1],
        ['Mali', 'typ1', datetime(2020,3,1), 'actor1', 1],
        ['Mali', 'typ2',datetime(2020,1,1), 'actor1', 0],
        ['Mali', 'typ2', datetime(2020,2,1), 'actor1', 0],
        ['Mali', 'typ1', datetime(2020,1,1), 'actor3', 0]],
        columns=['admin','FIVE_TYPE','EVENT_DATE','actor','FATALITIES',]))
    df =(df.sort(by=pl.col('EVENT_DATE'))
    .groupby_dynamic(
        index_column="EVENT_DATE",
        every="1mo",
        closed='left',
        period="2mo",
        offset= "-2mo",
        by=['admin', 'FIVE_TYPE'],
        include_boundaries=True,
    ).agg([
        pl.col('actor').filter(pl.col('FATALITIES') > 0).unique().alias('test2'),
    ]))
    print(df)
```


#### What is the actual behavior?

```
ctor_data_test.py:None (actor_data_test.py)
actor_data_test.py:70: in <module>
    test_get_new_actor_count()
actor_data_test.py:48: in test_get_new_actor_count
    df =(df.sort(by=pl.col('EVENT_DATE'))
/home/sebastian/airflow_env/lib/python3.8/site-packages/polars/internals/frame.py:4444: in agg
    self.df.lazy()
/home/sebastian/airflow_env/lib/python3.8/site-packages/polars/internals/lazy_frame.py:398: in collect
    return pli.wrap_df(ldf.collect())
E   pyo3_runtime.PanicException: implementation error groups are out of bounds; please open an issue
```

#### What is the expected behavior?

```
┌───────┬───────────┬─────────────────────┬─────────────────────┬─────────────────────┬────────────┐
│ admin ┆ FIVE_TYPE ┆ _lower_boundary     ┆ _upper_boundary     ┆ EVENT_DATE          ┆ test2      │
│ ---   ┆ ---       ┆ ---                 ┆ ---                 ┆ ---                 ┆ ---        │
│ str   ┆ str       ┆ datetime[ms]        ┆ datetime[ms]        ┆ datetime[ms]        ┆ list [str] │
╞═══════╪═══════════╪═════════════════════╪═════════════════════╪═════════════════════╪════════════╡
│ Mali  ┆ typ2      ┆ 2019-11-01 00:00:00 ┆ 2020-01-01 00:00:00 ┆ 2020-01-01 00:00:00 ┆ []         │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Mali  ┆ typ2      ┆ 2019-12-01 00:00:00 ┆ 2020-02-01 00:00:00 ┆ 2020-02-01 00:00:00 ┆ []         │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Mali  ┆ typ1      ┆ 2019-11-01 00:00:00 ┆ 2020-01-01 00:00:00 ┆ 2020-01-01 00:00:00 ┆ []         │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Mali  ┆ typ1      ┆ 2020-01-01 00:00:00 ┆ 2020-03-01 00:00:00 ┆ 2020-03-01 00:00:00 ┆ []         │
└───────┴───────────┴─────────────────────┴─────────────────────┴─────────────────────┴────────────┘
```

--- a/polars/polars-lazy/src/physical_plan/expressions/mod.rs
+++ b/polars/polars-lazy/src/physical_plan/expressions/mod.rs
@@ -361,10 +361,12 @@ impl<'a> AggregationContext<'a> {
                 // because this is lazy, we first must to update the groups
                 // by calling .groups()
                 self.groups();
-                assert!(
-                    self.groups.len() <= s.len(),
-                    "implementation error groups are out of bounds; please open an issue"
-                );
+                #[cfg(debug_assertions)]
+                {
+                    if self.groups.len() > s.len() {
+                        eprintln!("groups may be out of bounds; more groups than elements in a series is only possible in dynamic groupby")
+                    }
+                }
 
                 let out = s
                     .agg_list(&self.groups)

